Fix #10764: delete url rewrites for products after an attribute set is deleted

patrikpihlstrom opened the pull request:
### Description
Adds an interceptor around the delete function of `\Magento\Eav\Model\AttributeSetRepository` in order to capture the relevant product ids, and then remove the old url rewrites after the attribute set has been deleted.

### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#10764: Delete Attribute Set is not Deleting product url key in url rewrite

### Manual testing scenarios
1. Create an attribute set
2. Create a product from the newly created attribute set and insure that a url key is specified
3. Delete the attribute set
4. Create a new product from the default attribute set and apply the same url key
5. Hopefully you won't get an error when you save the product

Commit:
Fix depends field not working for radio elements

Fixes #9360

vrann commented on app/code/Magento/CatalogUrlRewrite/Plugin/Eav/Model/AttributeSetRepository.php:
```diff
@@ -0,0 +1,71 @@
+ * Time: 22:12
+ */
+
+namespace Magento\CatalogUrlRewrite\Plugin\Eav\Model;
```
Please fix the namespace. Here is the test result:

```Magento\Test\Integrity\ClassesTest::testClassReferences
Passed: 24399, Failed: 1, Incomplete: 0, Skipped: 0.
Data set: /opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/app/code/Magento/CatalogUrlRewrite/Plugin/Eav/Model/AttributeSetRepository.php
Incorrect namespace usage(s) found in file /opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/app/code/Magento/CatalogUrlRewrite/Plugin/Eav/Model/AttributeSetRepository.php:
CouldNotDeleteException
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/dev/tests/static/testsuite/Magento/Test/Integrity/ClassesTest.php:598
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/dev/tests/static/testsuite/Magento/Test/Integrity/ClassesTest.php:371
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:56
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/dev/tests/static/testsuite/Magento/Test/Integrity/ClassesTest.php:373

/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:115
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/lib/internal/Magento/Framework/App/Utility/AggregateInvoker.php:70
/opt/bamboo/bamboo-agent-home/xml-data/build-dir/MCCE22-AT398-INSE/magento2ce/dev/tests/static/testsuite/Magento/Test/Integrity/ClassesTest.php:373```

patrikpihlstrom replied:
I'm not quite sure what the namespace is supposed to be if not `Magento\CatalogUrlRewrite\Plugin\Eav\Model`. I've copied the namespace structure from other plugins, for example `Magento\Catalog\Plugin\Model\ResourceModel`.

vrann commented on app/code/Magento/CatalogUrlRewrite/Plugin/Eav/Model/AttributeSetRepository.php:
```diff
@@ -0,0 +1,71 @@
+        callable $proceed,
+        AttributeSetInterface $attributeSet
+    ) {
+        if (!isset($subject) || !$attributeSet->getAttributeSetId())
```
please fix the coding style

```
Magento\Test\Php\LiveCodeTest::testCodeStyle
PHP Code Sniffer detected 2 violation(s): 

FILE: ...to/CatalogUrlRewrite/Plugin/Eav/Model/AttributeSetRepository.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 50 | ERROR | [x] Expected 1 space after closing parenthesis; found 9
----------------------------------------------------------------------
```

vrann commented:
@patrikpihlstrom please make needed changes in order to accept the PR

YevSent committed:
Potential error on order edit page when address has extension attributes

 - Changed shipping and billing address comparision
 - Refactored related tests

RomaKis committed:
7691: address with saveInAddressBook 0 are still being added to the address book for new customers(backport to 2.2)

nmalevanec committed:
8003: Using System Value for Base Currency Results in Config Error.

Commit:
MAGETWO-83547: update button.phtml overcomplicated translation phrase. 2.2 #12136

Commit:
MAGETWO-83490: Cleanup for object manager references and depricated method #12061

Commit:
MAGETWO-83310: Fix bug: Customer import deletes exiting customer entity Fields #11968

Commit:
MAGETWO-82292: close #10810 Migrates Apache Access Syntax to 2.4 on Apache >= 2.4 #11459

Commit:
MAGETWO-83495: After logging in customer is now not redirecting to Customer Dashboard by default #11876

 - Merge Pull Request magento/magento2#11876 from p-bystritsky/magento2:ISSUE-10834
 - Merged commits:
   1. fe1b4b19192ae64e7c83939e8f9fe89fbba593ea
   2. 72babf2a17926d2daca12184b63a7e7b7f2cdf72
   3. 8e4bd2c1290430675b0353c88c911e800dd97f85
   4. c50899e50320a0bf20892ce4c3022523eae73928

Commit:
Merge pull request #1702 from magento-partners/prs-2.2-2017-11-10

[EngCom] Partners Pull Requests

Commit:
Merge pull request #1704 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12136 update button.phtml overcomplicated translation phrase. 2.2 by @ChuckyK
magento/magento2#12061 Cleanup for object manager references and depricated method by @atishgoswami
magento/magento2#11968 Fix bug: Customer import deletes exiting customer entity Fields by @jalogut
magento/magento2#11459 close #10810 Migrates Apache Access Syntax to 2.4 on Apache >= 2.4 by @jonashrem

Fixed Public Issues

magento/magento2#10810 Add support of apache2.4 commands in htaccess

Commit:
MAGETWO-83184: Static Content Update for ReleaseNotification Module

Commit:
MAGETWO-83495: After logging in customer is now not redirecting to Customer Dashboard by default #11876

Commit:
MAGETWO-83184: Static Content Update for ReleaseNotification Module

Commit:
MAGETWO-83184: Static Content Update for ReleaseNotification Module

Commit:
Merge pull request #1705 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#11876 After logging in customer is now not redirecting to Customer Dashboard by default by @p-bystritsky

Fixed Public Issues

magento/magento2#10834 signing in after selecting checkout button, will not end up to checkout page!

andrew-garside-temando committed:
Update the fixture and tests to reflect the functionality of getShippingMethod

Commit:
Merge remote-tracking branch 'upstream/2.2-develop' into MAGETWO-83184-release-notification-content-update

Commit:
MAGETWO-72138: Upgrade outdated libraries w/composer (minor versions) - spike
- Merge branch '2.2-develop' into bugfixes

Commit:
Merge branch '2.2-develop' of https://github.com/magento-qwerty/magento2ce into MAGETWO-70726

Commit:
MAGETWO-70726: [GITHUB] [2.1] Store View Language switch leads to 404 on some cases #5416

Commit:
Merge pull request #1701 from magento-plankton/bugfixes

[Plankton] 2.2.2 Fixes

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into PR_in_2.2

ishakhsuvarov committed:
MAGETWO-83740: Even existing credit memos should be refundable if their state is open #11550

 - Merge Pull Request magento/magento2#11550 from ajpevers/magento2:patch-9
 - Merged commits:
   1. 9380c95ba8a12eac0f57b92a682b932b1ceb6a1f
   2. 8a6c3f2d833b9ec4a9cea76c3926b1a193a038f7
   3. 480f8ae7f4948918cffcf9d63838a584dce9b5b4

dmanners committed:
MAGETWO-83745: 11832: Create order (on Customer edit page) - not working from admin environment #11952

 - Merge Pull Request magento/magento2#11952 from RomaKis/magento2:11832
 - Merged commits:
   1. eeb7dffa25bca4100753820b31ab4b544bfdb615

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into PR_in_2.2

Commit:
Merge pull request #1703 from magento-chaika/PR_in_2.2

[Chaika] bugfixes 2.2

nmalevanec committed:
8003: Using System Value for Base Currency Results in Config Error.

interio-zz committed:
MAGETWO-52974: Configurable product options not saved when editing - for mainline

Commit:
Merge branch 'MAGETWO-70726' into QwertyPR20171107

Commit:
Merge branch 'MAGETWO-52974-3' into QwertyPR20171107

interio-zz committed:
MAGETWO-52974: Configurable product options not saved when editing - for mainline

RomaKis committed:
11740: Sending emails from Admin in Multi-Store Environment defaults to Primary Store

RomaKis committed:
Merge branch '2.2-develop' into 11740

Commit:
Merge branch 'MAGETWO-52974-3' into QwertyPR20171107

RomaKis committed:
11740: Sending emails from Admin in Multi-Store Environment defaults to Primary Store

Commit:
Merge remote-tracking branch 'upstream/2.2-develop' into MAGETWO-83184-release-notification-content-update

Commit:
MAGETWO-83184: Static Content Update for ReleaseNotification Module

Commit:
MAGETWO-83745: 11832: Create order (on Customer edit page) - not working from admin environment #11952

Commit:
MAGETWO-83682: Fix js error when disable/enable wysiwyg editor #12141

Commit:
MAGETWO-83672: [Backport 2.2] MAGETWO-71697: Fix possible bug when saving address with empty street line #10582 #12130

Commit:
MAGETWO-83668: 8022: Uncaught Error: Call to a member function addItem() on array in app/code/Magento/Sales/Model/Order/Shipment.php(backport to 2.2) #12173

Commit:
MAGETWO-83627: 11532: Duplicate Simple Product Throws Error: Undefined offset: 0 in SaveHandler.php on line 122 #12001

Commit:
MAGETWO-83477: 10628: Color attribute swatches are not visible if sorting is enabled #12077

Commit:
MAGETWO-83286: Fix newsletter subscriptions between stores #12035

Commit:
MAGETWO-81161: Fix #10477 Check cart rule subselect conditions against quote item children too #11274

Commit:
MAGETWO-83184: Static Content Update for ReleaseNotification Module

Commit:
Merge pull request #1709 from magento-qwerty/QwertyPR20171107

Bug

MAGETWO-70726 [GITHUB] [2.1] Store View Language switch leads to 404 on some cases #5416
MAGETWO-52974 Configurable product options not saved when editing - for mainline

Commit:
Merge remote-tracking branch 'upstream/2.2-develop' into MAGETWO-83184-release-notification-content-update

Commit:
Merge pull request #1710 from magento-arcticfoxes/MAGETWO-83184-release-notification-content-update

[arcticfoxes] MAGETWO-83184: Static Content Update for ReleaseNotification Module

magento-team committed:
[EngCom] Public Pull Requests - 2.2-develop \n\n - merged latest code from mainline branch

Commit:
MAGETWO-83780: 7691: address with saveInAddressBook 0 are still being added to the address book for new customers(backport to 2.2) #12171

 - Merge Pull Request magento/magento2#12171 from RomaKis/magento2:7691-2.2
 - Merged commits:
   1. b750e583b36d4409dfd2b2b3a82bea06a31b2511

Commit:
Merge pull request #1711 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#11952 11832: Create order (on Customer edit page) - not working from admin environment by @RomaKis
magento/magento2#12141 Fix js error when disable/enable wysiwyg editor by @vovayatsyuk
magento/magento2#12130 [Backport 2.2] MAGETWO-71697: Fix possible bug when saving address with empty street line #10582 by @vovayatsyuk
magento/magento2#12173 8022: Uncaught Error: Call to a member function addItem() on array in app/code/Magento/Sales/Model/Order/Shipment.php(backport to 2.2) by @RomaKis
magento/magento2#12001 11532: Duplicate Simple Product Throws Error: Undefined offset: 0 in SaveHandler.php on line 122 by @RomaKis
magento/magento2#12077 10628: Color attribute swatches are not visible if sorting is enabled by @RomaKis
magento/magento2#12035 Fix newsletter subscriptions between stores by @sbaixauli
magento/magento2#11274 Fix #10477 Check cart rule subselect conditions against quote item children too by @marinagociu

Fixed Public Issues

magento/magento2#11832 Create order (on Customer edit page) - not working from admin environment
magento/magento2#8022 Uncaught Error: Call to a member function addItem() on array in app/code/Magento/Sales/Model/Order/Shipment.php
magento/magento2#11532 Duplicate Simple Product Throws Error: Undefined offset: 0 in SaveHandler.php on line 122
magento/magento2#10628 Color attribute swatches are not visible if sorting is enabled
magento/magento2#10014 Newsletter subscriptions status not isolated between multi stores
magento/magento2#10477 Cart price rule has failed if use dropdown attribute

kristof-ringleff committed:
MAGETWO-83783: Shipping method fixtures not compatible with getShippingMethod(true) in OrderCreateTest #12227

 - Merge Pull Request magento/magento2#12227 from andrew-garside-temando/magento2:2.2-develop
 - Merged commits:
   1. 149268c5c30267a04d50f8cc63f2ea3df0b8708e

Commit:
use InvoiceInterface instead of Invoice model to check if an invoice is set on the credit memo

nmalevanec committed:
8003: Using System Value for Base Currency Results in Config Error.

vovayatsyuk committed:
Fixed php notice when invalid ui_component config is used

RomaKis committed:
10128: New Orders not being saved to order grid

RomaKis committed:
Merge branch '2.2-develop' into 10128

ishakhsuvarov committed:
MAGETWO-83815: Fixed php notice when invalid ui_component config is used #12239

 - Merge Pull Request magento/magento2#12239 from vovayatsyuk/magento2:fix-undefined-offset-notice
 - Merged commits:
   1. 7f6b56078edb8eacc5b54e41cc84deb2fcf1fb41

Commit:
Backward compatible constructor updates

* Make new properties `private`
* Update new constructor arguments to follow Backward compatible
development guide

RomaKis committed:
10128: New Orders not being saved to order grid

Commit:
Make new test class properties `private`

RomaKis committed:
10128: New Orders not being saved to order grid

Commit:
New validation: validate-no-utf8mb4-characters.

danielrenaud committed:
MAGETWO-80622: СheckCustomerTypeTest failure on FAT-EXT-B2B-UR

Commit:
MAGETWO-83780: 7691: address with saveInAddressBook 0 are still being added to the address book for new customers(backport to 2.2) #12171

Commit:
Fix PHPDoc for uploadMediaFiles function in order to show $renameFileOff
param as optional set as false by default.

Commit:
MAGETWO-83868: 11941: Invoice for products that use qty decimal rounds down to whole number. #11997

 - Merge Pull Request magento/magento2#11997 from nmalevanec/magento2:11941
 - Merged commits:
   1. 0db2b08cb125046077b0abf1039718a3f6805cf9

Commit:
Fix PHPDoc typo on __construct method.

Commit:
Fix PHPDoc _saveProducts method to include throws
\Magento\Framework\Exception\LocalizedException

Commit:
Remove deprecated gmDate function from _saveProducts() and _saveStockItem()
functions. Used Intl library for datetime handling:
http://php.net/manual/en/book.intl.php

Commit:
Fix PHPDoc for _saveValidatedBunches(). Return shall be declared as
$this|\Magento\ImportExport\Model\Import\Entity\AbstractEntity
Removed 'void' as declaration from _saveValidatedBunches() function as
it shall return $this [\Magento\ImportExport\Model\Import\Entity\AbstractEntity]

Commit:
Fine tuning: add use Magento\Framework\Exception\LocalizedException in
order to use LocalizedException and reuse ObjectManager instead of classpath.

Commit:
Fine tuning: add use Magento\Framework\Exception\LocalizedException in
order to use LocalizedException.

Commit:
Merge pull request #1719 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12171 7691: address with saveInAddressBook 0 are still being added to the address book for new customers(backport to 2.2) by @RomaKis

Fixed Public Issues

magento/magento2#7691 address with saveInAddressBook 0 are still being added to the address book for new customers

Commit:
MAGETWO-83815: Fixed php notice when invalid ui_component config is used #12239

toddbc committed:
Always use linkIds for category flat EAV indexing.

This keeps full and rows indexing consistent.

folektoras committed:
Merge pull request #1 from mediact/bugfix/remove-reference-container

Update wishlist_email_rss.xml

ishakhsuvarov committed:
Merge pull request #1720 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-83815: Fixed php notice when invalid ui_component config is used #12239

p-bystritsky committed:
magento/magento2#12083: Cannot import zero (0) value into custom attribute

p-bystritsky committed:
magento/magento2#12083: Cannot import zero (0) value into custom attribute

RomaKis committed:
9764: exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder

ishakhsuvarov committed:
magento/magento2#11670: Merge branch '2.2-develop' of github.com:magento/magento2 into issue-11669

ishakhsuvarov committed:
MAGETWO-83552: save invoice ID on credit memo when using API method salesRefundInvoiceV1 #11670

 - Merge Pull Request magento/magento2#11670 from ajpevers/magento2:issue-11669
 - Merged commits:
   1. cfd92c4ff13bcd55d0ac6b06a820d74ee78f2558
   2. 94ae23d423c1647e2d9722f6a5cdd42c9bb7902d
   3. 914520985b2ed374a57dd74df8688fff46f06e5b
   4. ce20c0a5a9b27ace8d9891e38e864ffecadb2eff
   5. 23f315c82601a3b89499b7a288833f5d36c798a2
   6. 72533ad5801b3839398e1bd091495537a5429361
   7. 1d341efa8875779b266c00bee84c9d88f2037c59
   8. f2ba426ebf75937a3390c8166a08323128a302f4

Commit:
MAGETWO-81841: Added CLI command to enable and disable the Profiler #11407

slavvka committed:
Merge remote-tracking branch 'origin/MAGETWO-80622' into PR2.2_20171115

Commit:
MAGETWO-81841: Added CLI command to enable and disable the Profiler #11407

Commit:
Merge pull request #1726 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#11407 Added CLI command to enable and disable the Profiler by @peterjaap

Fixed Public Issues

magento/magento2#9277 Create new CLI command: enable/disable Magento Profiler

Commit:
MAGETWO-83957: magento/magento2#12083: Cannot import zero (0) value into custom attribute #12283

 - Merge Pull Request magento/magento2#12283 from p-bystritsky/magento2:ISSUE-12083
 - Merged commits:
   1. 41e9ec05e7b7330ff0ee0d278e1012527d9889b5
   2. 89d613c1da1c7a054fc256df3fda206ff32bd458

Commit:
MAGETWO-83958: Phpdoc improvements #12257

 - Merge Pull Request magento/magento2#12257 from KarlDeux/magento2:feature/phpdoc-improvements
 - Merged commits:
   1. f9366049ae72400c587a6fe7225d9c6022bc1218
   2. 3ce246d476e090cb45b91ffa45416a1b2aade7f0
   3. 8f1aa87d249294fd756469ccf4f84a271d21c432
   4. 41a43c1a57bc0a38ccce14f27e5ae8373d0fd76d
   5. 19de842b1d291f731937adcbe0c1aa3213cbb861
   6. dc22a8249dad15e168a781a2c32ffebdf5c51869
   7. 820c0973bfeb5ba67e374c367bb584261f2ac0c1

Commit:
MAGETWO-83958: Phpdoc improvements #12257

 - reverted some changes that were done incorrectly

Commit:
MAGETWO-83958: Phpdoc improvements #12257

madonzy committed:
Issue: 3596. Resolve Notice with undefined index 'value' while preview order with instance payable method selected if custom offline payment method is defined in system

Commit:
Merge pull request #1728 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12257 Phpdoc improvements by @KarlDeux

Commit:
MAGETWO-83959: Issue: 3596. Resolve Notice with undefined index 'value' #12296

 - Merge Pull Request magento/magento2#12296 from madonzy/magento2:hotfix/3596-undevined-index-value-in-order-view-2.2
 - Merged commits:
   1. addda4281d079e1b9988ee6fc97d7d79f092acfd

RomaKis committed:
9764: exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder

Commit:
MAGETWO-75217: SKU parameter in capital letters on CSV ignored by database in Import but OK'd by validator

Commit:
Merge branch '2.2-develop' of https://github.com/magento-qwerty/magento2ce into MAGETWO-75217

slavvka committed:
Merge pull request #1724 from magento-borg/PR2.2_20171115

Fixed issues:
- MAGETWO-80622 СheckCustomerTypeTest failure on FAT-EXT-B2B-UR

vovayatsyuk committed:
Fix errors when DOB field is visible. #12146

 - Fix customer create page rendering
 - Fix customer save in backend and frontend

Commit:
MAGETWO-75217: SKU parameter in capital letters on CSV ignored by database in Import but OK'd by validator

Commit:
Merge branch '2.2-develop' of https://github.com/magento-qwerty/magento2ce into MAGETWO-75217

Commit:
Handle empty or incorrect lines in a language CSV

ishakhsuvarov committed:
MAGETWO-83959: Issue: 3596. Resolve Notice with undefined index 'value' #12296

ishakhsuvarov committed:
MAGETWO-83957: magento/magento2#12083: Cannot import zero (0) value into custom attribute #12283

ishakhsuvarov committed:
MAGETWO-83868: 11941: Invoice for products that use qty decimal rounds down to whole number. #11997

RomaKis committed:
12261: Order confirmation email contains non functioning links #12261

ishakhsuvarov committed:
MAGETWO-83999: 12261: Order confirmation email contains non functioning links #12261 #12308

 - Merge Pull Request magento/magento2#12308 from RomaKis/magento2:12261
 - Merged commits:
   1. 24eee6414605d1b2ec0640fbfd9e8c660fbd960d
   2. ec33cc50bbe9be7559b179f4a0d75b4ca062ecb8

RomaKis committed:
12261: Order confirmation email contains non functioning links #12261

omiroshnichenko committed:
MAGETWO-83993: Fixed a js bug where ui_component labels have the wrong sort order. #11846

osrecio committed:
MAGETWO-84003: 9764: exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder #12303

 - Merge Pull Request magento/magento2#12303 from RomaKis/magento2:9764
 - Merged commits:
   1. ba47d85ec578e7635eb8db76d58d2467a5b52a4e
   2. 0a76034167a2f92e1d066014f899e899d2e42768

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into fix-ui-component-wrong-sort-order

Commit:
Merge github.com:magento/magento2 into develop

Commit:
change robots.txt response content type to 'text/plain'

ishakhsuvarov committed:
Merge pull request #1730 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-83959: Issue: 3596. Resolve Notice with undefined index 'value' #12296
 - MAGETWO-83957: magento/magento2#12083: Cannot import zero (0) value into custom attribute #12283
 - MAGETWO-83868: 11941: Invoice for products that use qty decimal rounds down to whole number. #11997

Commit:
MAGETWO-84016: Fixed 'Non-numeric value' warning on account create/save when DOB field is visible #12302

 - Merge Pull Request magento/magento2#12302 from vovayatsyuk/magento2:fix-customer-create-and-save-with-dob
 - Merged commits:
   1. 799199e149b749890a8ed15b27d751cc7c57dade

Commit:
MAGETWO-84003: 9764: exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder #12303

Commit:
Merge pull request #1733 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12303 9764: exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder by @RomaKis

Fixed Public Issues

magento/magento2#9764 exception message is wrong and misleading in findAccessorMethodName() of Magento\Framework\Reflection\NameFinder

dmanners committed:
MAGETWO-83329: 8003: Using System Value for Base Currency Results in Config Error. #11809

 - Merge Pull Request magento/magento2#11809 from nmalevanec/magento2:Github-8003
 - Merged commits:
   1. 38735ac4b3200e81c5e7f28beb4ec7f4b76916b8
   2. b63ceb99d650e23d62c6a03230f01ff7f85b6920
   3. 3e0c785820acd4bcae6f0b2c3ff1bc706bc10499

bartlubbersen committed:
[BUGFIX] All option values were deleted if one was deleted earlier

The problem occurs if you delete one custom option value was deleted. This occurred because the same value object is used for saving all values, and if one value is deleted the _isDeleted flag is set tot true on that object. So all values after that value will also be deleted.

RomaKis committed:
9684: No ACL set for integrations

Commit:
MAGETWO-75217: SKU parameter in capital letters on CSV ignored by database in Import but OK'd by validator

dmanners committed:
MAGETWO-84081: 9684: No ACL set for integrations #12332

 - Merge Pull Request magento/magento2#12332 from RomaKis/magento2:9684
 - Merged commits:
   1. ffb781737a927db44a925fd89dcf97101b7eb67b

Commit:
Merge magento-partners/magento2ce#80.

Commit:
Merge magento-partners/magento2ce#65.

Commit:
Merge magento-partners/magento2ce#74.

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-83402-PR-11592

ishakhsuvarov committed:
magento/magento2#12304: Merge branch '2.2-develop' of github.com:magento/magento2 into bugfix/language-pack-empty-line

ishakhsuvarov committed:
magento/magento2#12304: Handle empty or incorrect lines in language CSV
 - Updated test data to cover this case

ishakhsuvarov committed:
MAGETWO-83977: Handle empty or incorrect lines in a language CSV #12304

 - Merge Pull Request magento/magento2#12304 from FreekVandeursen/magento2:bugfix/language-pack-empty-line
 - Merged commits:
   1. 470c257f7a02e23956a9a82245f02d4d0f22fa20
   2. 3d1d02741240a6894537b955721f8e0073162e3b
   3. 723452c71eb6da31f6ac8f221f4c587ce855a309

ishakhsuvarov committed:
magento/magento2#12310: Merge branch '2.2-develop' of github.com:magento/magento2 into fix-robots-txt-content-type

ishakhsuvarov committed:
magento/magento2#12310: Fix robots.txt content type to 'text/plain'

ishakhsuvarov committed:
MAGETWO-84006: Fix robots.txt content type to 'text/plain' #12310

 - Merge Pull Request magento/magento2#12310 from tufahu/magento2:fix-robots-txt-content-type
 - Merged commits:
   1. 8fc99b164bb0717a631fe61d080ac24314d4bfd0
   2. 36dfdca9b74f9df5bb3ad1f71cc51e782f2af379
   3. 5c024025593a701df675842fe7136dbe23df59fb
   4. 2eada9caa499963cb9083d7d72602f149114dfc6
   5. e9986c5ac7abdca30c20b416913c9458d974e5ad

ishakhsuvarov committed:
MAGETWO-84098: Webshop throws an exception when sharing wishlist with RSS enabled #12276

 - Merge Pull Request magento/magento2#12276 from mediact/magento2:2.2-mediact-develop
 - Merged commits:
   1. d47a3fe1be687dc1f76b45fa3677a089ee799fd8
   2. 9f8c7d120a41c39688af1837d4689b5d8e1c68f9

Commit:
Stabilization magento-partners/magento2ce#83

ishakhsuvarov committed:
MAGETWO-84098: Webshop throws an exception when sharing wishlist with RSS enabled #12276

ishakhsuvarov committed:
MAGETWO-84081: 9684: No ACL set for integrations #12332

ishakhsuvarov committed:
MAGETWO-84006: Fix robots.txt content type to 'text/plain' #12310

ishakhsuvarov committed:
MAGETWO-83977: Handle empty or incorrect lines in a language CSV #12304

ishakhsuvarov committed:
Merge pull request #1734 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-84098: Webshop throws an exception when sharing wishlist with RSS enabled #12276
 - MAGETWO-84081: 9684: No ACL set for integrations #12332
 - MAGETWO-84006: Fix robots.txt content type to 'text/plain' #12310
 - MAGETWO-83977: Handle empty or incorrect lines in a language CSV #12304

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-83402-PR-11592

dmanners committed:
MAGETWO-82577: [Backport 2.2] Translate order getCreatedAtFormatted() to store locale #11422

 - Merge Pull Request magento/magento2#11422 from JeroenVanLeusden/magento2:patch-4
 - Merged commits:
   1. 9d0befb24b4dd81f1568155b23ec9451fea290b8
   2. 02bc89297aad77fbd68c6263d421a0d254af7a41

p-bystritsky committed:
magento/magento2#11691: Wrong return type for getAttributeText($attributeCode)

omiroshnichenko committed:
MAGETWO-83410: Fix #10438: Potential error on order edit page when address has extension attributes #11787

 - Merge Pull Request magento/magento2#11787 from joni-jones/magento2:G#10438
 - Merged commits:
   1. 2f0898a6fb8f619640b4dbc84fd5ab2a376dc0fa

slopukhov committed:
MAGETWO-83328: Run Nightly PAT on 2.2

Commit:
Merge branch magento/magento2/2.2-develop into magento-partners/magento2ce/prs-2.2-develop-2017-11-17

Commit:
Merge pull request #1735 from magento-partners/prs-2.2-develop-2017-11-17

[EngCom] Partners Pull Requests

Commit:
MAGETWO-77840: [2.2.x] - Special/lowest price in child of a Configurable Product causes the entire product to show that price

Commit:
MAGETWO-70725: Admin token does not expire after 'Admin Token Lifetime (hours)'

- fix

Commit:
MAGETWO-84018: magento/magento2#11691: Wrong return type for getAttributeText() #12003

 - Merge Pull Request magento/magento2#12003 from p-bystritsky/magento2:ISSUE-11691
 - Merged commits:
   1. bd9054a4891bf9e78fdeac54e3136e28df4758a0

Commit:
MAGETWO-84018: magento/magento2#11691: Wrong return type for getAttributeText($attributeCode) #12003

Commit:
MAGETWO-83999: 12261: Order confirmation email contains non functioning links #12261 #12308

Commit:
MAGETWO-83410: Fix #10438: Potential error on order edit page when address has extension attributes #11787

Commit:
Merge pull request #1738 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12003 magento/magento2#11691: Wrong return type for getAttributeText($attributeCode) by @p-bystritsky
magento/magento2#12308 12261: Order confirmation email contains non functioning links #12261 by @RomaKis
magento/magento2#11787 Fix #10438: Potential error on order edit page when address has extension attributes by @joni-jones

Fixed Public Issues

magento/magento2#11691 Wrong return type for getAttributeText($attributeCode)
magento/magento2#12261 Order confirmation email contains non functioning links
magento/magento2#10438 Potential error on order edit page when address has extension attributes

Commit:
MAGETWO-84016: Fixed 'Non-numeric value' warning on account create/save when DOB field is visible #12302

Commit:
Merge pull request #1739 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12302 Fixed 'Non-numeric value' warning on account create/save when DOB field is visible by @vovayatsyuk

Fixed Public Issues

magento/magento2#12146 Customer with empty "Date of Birth" cannot be saved

Commit:
use FQCN throughout the class

mariuscris committed:
use FQCN, revert doc parameter namespace

mariuscris committed:
import Config class

Commit:
MAGETWO-81525: Fix email not sent when sitemap generation has errors #11320

 - Merge Pull Request magento/magento2#11320 from marinagociu/magento2:fix-undefined-property-translateinline
 - Merged commits:
   1. 0888e8ce84d9f29c97e7c1997b7ff243990c5201
   2. 3fc3c1dc76bf643915b752dbe13b064e3cbc858e
   3. 3c3168bdfbe43379e485ff91e708a045dd767e67
   4. 86f1419e2a3ef386f1ebe3b68b23ab3050c2f7bb
   5. 16590f018ab3c110d6999ebb467cd18592287f1c

Commit:
MAGETWO-77840: [2.2.x] - Special/lowest price in child of a Configurable Product causes the entire product to show that price

gomencal committed:
add static and api-funcional tests for swatches

Commit:
Merge remote-tracking branch 'main/2.2-develop' into MAGETWO-70725

Commit:
MAGETWO-70725: Admin token does not expire after 'Admin Token Lifetime (hours)'

- fix static tests

mariuscris committed:
revert type parameter on method prepareSortableFieldsByCategory

Commit:
MAGETWO-70725: Admin token does not expire after 'Admin Token Lifetime (hours)'

- fix static tests

omiroshnichenko committed:
MAGETWO-83401: Fix depends field not working for radio elements #11539

 - Merge Pull Request magento/magento2#11539 from jahvi/magento2:fix/depends-not-working-radio-inputs
 - Merged commits:
   1. 0031a1c56b5277149e3cdc567251c19dbcf8df12

ishakhsuvarov committed:
MAGETWO-81942: Magento 2.2.0 A solution for Product Repeat Issue after filter on category listing page. #11429

 - Merge Pull Request magento/magento2#11429 from mayankzalavadia/magento2:patch-issue11139
 - Merged commits:
   1. 1d998f9c45856497332daddd930786ee42d458cb
   2. 906daf673c4dd55a7f0452fb1b58797abf217cc9

ishakhsuvarov committed:
MAGETWO-83740: Even existing credit memos should be refundable if their state is open #11550

ishakhsuvarov committed:
MAGETWO-83329: 8003: Using System Value for Base Currency Results in Config Error. #11809

ishakhsuvarov committed:
MAGETWO-81942: Magento 2.2.0 A solution for Product Repeat Issue after filter on category listing page. #11429

ishakhsuvarov committed:
MAGETWO-81525: Fix email not sent when sitemap generation has errors #11320

Commit:
Merge pull request #1743 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#11550 Even existing credit memos should be refundable if their state is open by @ajpevers
magento/magento2#11809 8003: Using System Value for Base Currency Results in Config Error. by @nmalevanec
magento/magento2#11429 Magento 2.2.0 A solution for Product Repeat Issue after filter on category listing page. by @mayankzalavadia
magento/magento2#11320 Fix email not sent when sitemap generation has errors by @marinagociu

Fixed Public Issues

magento/magento2#8003 Using System Value for Base Currency Results in Config Error
magento/magento2#11139 Product Repeat Isuue after filter on category listing page.
magento/magento2#10502 Fatal error: Call getTranslateInline of null when generating some sitemap with errors

Commit:
MAGETWO-83402: Fix issue #10347 - Wrong order tax amounts displayed when using specific tax configuration (2.2-develop) #11592

Commit:
MAGETWO-84267: #11528 can't save customizable options #12048

 - Merge Pull Request magento/magento2#12048 from luismiguelyangehuaman/magento2:PR#CanNotsaveCustomizableOptions
 - Merged commits:
   1. ed5d2f524410f34e632a9cec3ccc8348eb395420
   2. 1caffdc0ef7e9b006444afab8f0060fe8684e29b

Commit:
MAGETWO-84267: #11528 can't save customizable options #12048

 - fixed code style

Commit:
Merge pull request #1746 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#11592 Fix issue #10347 - Wrong order tax amounts displayed when using specific tax configuration (2.2-develop) by @PieterCappelle

Fixed Public Issues

magento/magento2#10347 Wrong order tax amounts displayed when using specific tax configuration

Commit:
CAPTCHA LABEL : Update the captcha labels and help text to refer to the contents of the captcha image

Commit:
MAGETWO-84272: Update AbstractBackend.php #12120

 - Merge Pull Request magento/magento2#12120 from hewersonfreitas/magento2:patch-1
 - Merged commits:
   1. dd40f2b799381aa8475815774344eb4771920bef

Commit:
MAGETWO-84272: Update AbstractBackend.php #12120

 - fixed failed tests

ishakhsuvarov committed:
MAGETWO-84284: Update CAPTCHA labels to reflect the symbols in the CAPTCHA image #12387

 - Merge Pull Request magento/magento2#12387 from RhodriOwainDavies/magento2:captcha_label
 - Merged commits:
   1. ff734598c86942dd98f55bbd9e0a30b361da26e6

RomaKis committed:
9515: South Korea Zip Code Validation incorrect

dmanners committed:
Merge remote-tracking branch 'mainline/2.2-develop' into 11792

dmanners committed:
MAGETWO-84087: 11792: Can't add customizable options to product #11965

 - Merge Pull Request magento/magento2#11965 from RomaKis/magento2:11792
 - Merged commits:
   1. 081a3b793be0b76eb0e8058d2e54efe17ddd5fb8
   2. 559020c9c45b272f83a27180aa70ccb248ad19a7
   3. 311d409504851505182c388b0428301d4bc9471c

dmanners committed:
Merge remote-tracking branch 'mainline/2.2-develop' into 12064

dmanners committed:
MAGETWO-84125: 12064: Database Rollback not working with magento 2.1.9? #12108

 - Merge Pull Request magento/magento2#12108 from RomaKis/magento2:12064
 - Merged commits:
   1. 66677ae715a7899a1c3f88ccf6e02a902757bfca
   2. 74980f2aea09799e3ac81cd196a2d66b9bcca29d

omiroshnichenko committed:
MAGETWO-83399: Fix for remove 'product_list_toolbar' block from layout in XML #9413 #11473

 - Merge Pull Request magento/magento2#11473 from mariuscris/magento2:issue-9413
 - Merged commits:
   1. 42072dd20ac633ce678946586de6d73b83e9a43b
   2. 9dbe44493b92407c9b02217966224cb021175c48
   3. 49fdfaee640a79ef6278e9df247a07217d9d597c
   4. b6ed2d9b31de64b234540abfd28f55485c4556bd
   5. 9cd20c03f9c4f3d7dd79865addf8dac9aa6139db
   6. 2adefdff73c62d829d1969b3abc988992bd3da39
   7. cbda7b821c976d82eb3c513c435cd941e4b287fb
   8. 992b3a10abfa7ad39a4f263f61d1b47f36113ae0
   9. 4e937a39fa59a7cc0cefcd19691bc095accc1c9e
   10. c303e4dcbc0501a125e844d521d5fc57624b6b8b
   11. cd7dc5652b56b62713808233b0220cc7c8741078

slopukhov committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into MAGETWO-83328

Commit:
MAGETWO-84311: Fix for issue 12127: Single quotation marks are now decoded properly in admin attribute option input fields #12133

 - Merge Pull Request magento/magento2#12133 from erfanimani/magento2:2.2-develop
 - Merged commits:
   1. cfab448cc75e36a1bbfe4af5eab476b495e33ae3

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into issue-9413

vzabaznov committed:
Merge pull request #1748 from magento-performance/MAGETWO-83328

[Performance] MAGETWO-83328: Run Nightly PAT on 2.2

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-84311-magento-magento2-12133

Commit:
Merge pull request #2 from magento/2.2-develop

PR2

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into fix-ui-component-wrong-sort-order

omiroshnichenko committed:
Merge branch '2.2-develop' of github.com:magento/magento2 into update-readme-with-labels

omiroshnichenko committed:
MAGETWO-83621: Add link to issue gates wiki page in the labels section of the readme #12154

 - Merge Pull Request magento/magento2#12154 from dmanners/magento2:update-readme-with-labels
 - Merged commits:
   1. ff249e1dfbe355dbe3d55e050fefe2dc6b63dfb5
   2. 1d16c6d3443ba1b63e2320501aa1426f7daf7791

RomaKis committed:
9515: South Korea Zip Code Validation incorrect

ishakhsuvarov committed:
MAGETWO-83993: Fixed a js bug where ui_component labels have the wrong sort order. #11846

 - Merge Pull Request magento/magento2#11846 from deiserh/magento2:fix-ui-component-wrong-sort-order
 - Merged commits:
   1. 0f948ac1403ecbdacae6c58cfdd5b045698c1ea8
   2. 6f70d065a415157a57716e8629b523de0ec6f291
   3. 8f52242e060aefedc0bf35264d4da437870168db
   4. c464951669e729d163c05a0955869cf4278a8106

ishakhsuvarov committed:
MAGETWO-84284: Update CAPTCHA labels to reflect the symbols in the CAPTCHA image #12387

ishakhsuvarov committed:
MAGETWO-84267: #11528 can't save customizable options #12048

ishakhsuvarov committed:
MAGETWO-84125: 12064: Database Rollback not working with magento 2.1.9? #12108

ishakhsuvarov committed:
MAGETWO-84087: 11792: Can't add customizable options to product #11965

ishakhsuvarov committed:
MAGETWO-83993: Fixed a js bug where ui_component labels have the wrong sort order. #11846

ishakhsuvarov committed:
MAGETWO-83401: Fix depends field not working for radio elements #11539

Commit:
Merge pull request #1750 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12387 Update CAPTCHA labels to reflect the symbols in the CAPTCHA image by @RhodriOwainDavies
magento/magento2#12048 #11528 can't save customizable options by @luismiguelyangehuaman
magento/magento2#12108 12064: Database Rollback not working with magento 2.1.9? by @RomaKis
magento/magento2#11965 11792: Can't add customizable options to product by @RomaKis
magento/magento2#11846 Fixed a js bug where ui_component labels have the wrong sort order. by @deiserh
magento/magento2#11539 Fix depends field not working for radio elements by @jahvi

Fixed Public Issues

magento/magento2#11528 Validation prevents form closing
magento/magento2#12064 Database Rollback not working with magento 2.1.9?
magento/magento2#11792 Can't add customizable options to product
magento/magento2#9360 field doesn't work in system.xml for "radios" fields

Commit:
MAGETWO-84272: Update AbstractBackend.php #12120

Commit:
MAGETWO-83621: Add link to issue gates wiki page in the labels section of the readme #12154

Commit:
MAGETWO-84371: 10210: Transport variable can not be altered in email_invoice_set_template_vars_before Event (backport MAGETWO-69482 to 2.2) #12132

 - Merge Pull Request magento/magento2#12132 from RomaKis/magento2:10210
 - Merged commits:
   1. 279b0c8e9f3a6ce90a71da3ce2642e122558d1aa

Commit:
Merge pull request #1754 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12120 Update AbstractBackend.php by @hewersonfreitas
magento/magento2#12154 Add link to issue gates wiki page in the labels section of the readme by @dmanners

Commit:
Merge branch 'MAGETWO-77840' into PR-22-2017-11-23

Commit:
Merge branch 'MAGETWO-75217' into PR-22-2017-11-23

Commit:
Merge branch 'MAGETWO-70725' into PR-22-2017-11-23

Commit:
MAGETWO-77840: [2.2.x] - Special/lowest price in child of a Configurable Product causes the entire product to show that price

- fixing FAT

gomencal committed:
drop useless fixture, fix tests

gomencal committed:
codestyle fix

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-83783-magento-magento2-12227

ishakhsuvarov committed:
magento/magento2#12241: Merge branch '2.2-develop' of github.com:magento/magento2 into 10128

ishakhsuvarov committed:
MAGETWO-84474: 10128: New Orders not being saved to order grid #12241

 - Merge Pull Request magento/magento2#12241 from RomaKis/magento2:10128
 - Merged commits:
   1. ef165a2d3666d92a109252541c399e55695c830b
   2. f11c0db40a86cde5f9e5ed74eeaccfa45e71d2d2
   3. 1ebcd7add6e6bcb2f8deb62ba5754e3110b59f12
   4. a933cf745b4e6cc82d07cefb00cd4af9874e034f

Commit:
MAGETWO-83290: Add swatch option: Prevent loosing data and default value if data is not populated via adminhtml #12036

 - Merge Pull Request magento/magento2#12036 from gomencal/magento2:10707_22
 - Merged commits:
   1. 6d5c1f5414af9fc44868913ae484596c2fc04bc8
   2. 1053d6c0b43ae8453b14ad678de19d8f02e9b76d
   3. 0599c8eb227536517ca3acbb4349695cb4f023b9
   4. 243dccba6ecde194f56c052ff91f2bdd641def09

ishakhsuvarov committed:
MAGETWO-83741: 11740: Sending emails from Admin in Multi-Store Environment defaults to Primary Store #11992

 - Merge Pull Request magento/magento2#11992 from RomaKis/magento2:11740
 - Merged commits:
   1. 9193745655b641d142d487c5e88f1a839cff821e
   2. ef03b523315c07c7cd26a5292885cf668a0b6e3d
   3. f2bfdd941c0c8c6c7c745156d1ce206e38c76820
   4. b6764c81aded90f2acb152c03e3750c3177b1187
   5. b3227183b97f6ba993e2ee64b504f0ab98262e8b

ishakhsuvarov committed:
MAGETWO-83552: save invoice ID on credit memo when using API method salesRefundInvoiceV1 #11670

ishakhsuvarov committed:
MAGETWO-82577: [Backport 2.2] Translate order getCreatedAtFormatted() to store locale #11422

orlangur committed:
MAGETWO-84367: New validation: 3bytes characters filter (4 bytes characters cannot be stored using UTF8) #12253

 - Merge Pull Request magento/magento2#12253 from KarlDeux/magento2:antiemoji
 - Merged commits:
   1. b64e4703de287e65b12e5c7f280643b1d1377808
   2. c21229cb8e91ebd30a28ad63aa87405faeafeee6

ishakhsuvarov committed:
MAGETWO-84474: 10128: New Orders not being saved to order grid #12241

ishakhsuvarov committed:
MAGETWO-83783: Shipping method fixtures not compatible with getShippingMethod(true) in OrderCreateTest #12227

ishakhsuvarov committed:
MAGETWO-83290: Add swatch option: Prevent loosing data and default value if data is not populated via adminhtml #12036

dvoskoboinikov committed:
Merge pull request #1759 from magento-qwerty/PR-22-2017-11-23

Fixed issues:
- MAGETWO-75217: SKU parameter in capital letters on CSV ignored by database in Import but OK'd by validator
- MAGETWO-77840: [2.2.x] - Special/lowest price in child of a Configurable Product causes the entire product to show that price
- MAGETWO-70725: Admin token does not expire after 'Admin Token Lifetime (hours)'

ishakhsuvarov committed:
MAGETWO-83741: 11740: Sending emails from Admin in Multi-Store Environment defaults to Primary Store #11992

ishakhsuvarov committed:
MAGETWO-83399: Fix for remove 'product_list_toolbar' block from layout in XML #9413 #11473

ishakhsuvarov committed:
MAGETWO-83399: Fix for remove 'product_list_toolbar' block from layout in XML #9413 #11473

 - Merge Pull Request magento/magento2#11473 from mariuscris/magento2:issue-9413
 - Merged commits:
   1. 42072dd20ac633ce678946586de6d73b83e9a43b
   2. 9dbe44493b92407c9b02217966224cb021175c48
   3. 49fdfaee640a79ef6278e9df247a07217d9d597c
   4. b6ed2d9b31de64b234540abfd28f55485c4556bd
   5. 9cd20c03f9c4f3d7dd79865addf8dac9aa6139db
   6. 2adefdff73c62d829d1969b3abc988992bd3da39
   7. cbda7b821c976d82eb3c513c435cd941e4b287fb
   8. 992b3a10abfa7ad39a4f263f61d1b47f36113ae0
   9. 4e937a39fa59a7cc0cefcd19691bc095accc1c9e
   10. c303e4dcbc0501a125e844d521d5fc57624b6b8b
   11. cd7dc5652b56b62713808233b0220cc7c8741078
   12. 616246465e88a83ba815d163d5c86e72cb9dd630

ishakhsuvarov committed:
Merge branch 'MAGETWO-83399-magento-magento2-11473' of github.com:magento-engcom/magento2ce into 2.2-develop-prs

ishakhsuvarov committed:
Merge branch '2.2-develop' of github.com:magento/magento2ce into 2.2-develop-prs

ishakhsuvarov committed:
Merge pull request #1764 from magento-engcom/2.2-develop-prs

[EngCom] Public Pull Requests - 2.2-develop
 - MAGETWO-83552: save invoice ID on credit memo when using API method salesRefundInvoiceV1 #11670
 - MAGETWO-82577: [Backport 2.2] Translate order getCreatedAtFormatted() to store locale #11422
 - MAGETWO-84474: 10128: New Orders not being saved to order grid #12241
 - MAGETWO-83783: Shipping method fixtures not compatible with getShippingMethod(true) in OrderCreateTest #12227
 - MAGETWO-83290: Add swatch option: Prevent loosing data and default value if data is not populated via adminhtml #12036
 - MAGETWO-83741: 11740: Sending emails from Admin in Multi-Store Environment defaults to Primary Store #11992
 - MAGETWO-83399: Fix for remove 'product_list_toolbar' block from layout in XML #9413 #11473

RomaKis committed:
9742: Default welcome message returns after being deleted #9742

Commit:
MAGETWO-84372: 9515: South Korea Zip Code Validation incorrect #903

 - Merge Pull Request magento-engcom/magento2ce#903 from RomaKis/magento2:9515
 - Merged commits:
   1. 97d449ce3b90a18a6292c2c739cb325519769985
   2. 5c3d29503b89d7b5f2623e33d1798c7c25dd6dcd

Commit:
MAGETWO-84372: 9515: South Korea Zip Code Validation incorrect #903

Commit:
MAGETWO-84371: 10210: Transport variable can not be altered in email_invoice_set_template_vars_before Event (backport MAGETWO-69482 to 2.2) #12132

orlangur committed:
MAGETWO-81916: Attribute category_ids issue #11389

 - Merge Pull Request magento/magento2#11389 from manuelson/magento2:feature/11341
 - Merged commits:
   1. c33d9d7017c1067b3e8122ceb25741eda0858ea8

Commit:
Merge pull request #1769 from magento-engcom/2.2-develop-prs

Public Pull Requests

#1770 9515: South Korea Zip Code Validation incorrect by @RomaKis
magento/magento2#12132 10210: Transport variable can not be altered in email_invoice_set_template_vars_before Event (backport MAGETWO-69482 to 2.2) by @RomaKis

Fixed Public Issues

magento/magento2#9515 South Korea Zip Code Validation incorrect
magento/magento2#10210 Transport variable can not be altered in email_invoice_set_template_vars_before Event

Commit:
Merge remote-tracking branch 'mainline/2.2-develop' into MAGETWO-84311-magento-magento2-12133

Commit:
MAGETWO-84413: 9742: Default welcome message returns after being deleted #9742 #12328

 - Merge Pull Request magento/magento2#12328 from RomaKis/magento2:9742
 - Merged commits:
   1. 147f6380f77ce76d2616ae40bff278215ca8d27a

Commit:
MAGETWO-84413: 9742: Default welcome message returns after being deleted #9742 #12328

Commit:
MAGETWO-84367: New validation: 3bytes characters filter (4 bytes characters cannot be stored using UTF8) #12253

Commit:
MAGETWO-84311: Fix for issue 12127: Single quotation marks are now decoded properly in admin attribute option input fields #12133

Commit:
MAGETWO-81916: Attribute category_ids issue #11389

Commit:
Merge pull request #1773 from magento-engcom/2.2-develop-prs

Public Pull Requests

magento/magento2#12328 9742: Default welcome message returns after being deleted #9742 by @RomaKis
magento/magento2#12253 New validation: 3bytes characters filter (4 bytes characters cannot be stored using UTF8) by @KarlDeux
magento/magento2#12133 Fix for issue 12127: Single quotation marks are now decoded properly in admin attribute option input fields by @erfanimani
magento/magento2#11389 Attribute category_ids issue by @manuelson

Fixed Public Issues

magento/magento2#9742 Default welcome message returns after being deleted
magento/magento2#12058 Can't save emoji in custom product options
magento/magento2#12127 Apostrophe in attribute option value in admin is not handled properly
magento/magento2#11341 Attribute category_ids issue

Commit:
MAGETWO-82668: Fix #10764: delete url rewrites for products after an attribute set is deleted #11391

 - Merge Pull Request magento/magento2#11391 from patrikpihlstrom/magento2:delete-url-rewrites
 - Merged commits:
   1. 284f966dc87dce72c238b0e3af3ccb72bfed84c4
   2. 07a95ccfc3f42e88f3d2ddcef7bb1c9ce4c3e2ae
   3. 58b95365ce231e92570b8214e970bc67d91583af
   4. 5ff6b675dcf6ba55fdab46a6afa402c40b04370c
   5. 4b95a1192ea208306f26e9ac31b847dca44c8cc0

Commit:
MAGETWO-82668: Fix #10764: delete url rewrites for products after an attribute set is deleted #11391

 - fixed static tests

Commit:
MAGETWO-82668: Fix #10764: delete url rewrites for products after an attribute set is deleted #11391

 - fixed plugin logic

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/not_signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=11391) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/magento2?pullRequest=11391) before we can accept your contribution.<br/>**6** out of **13** committers have signed the CLA.<br/><br/>:white_check_mark: okorshenko<br/>:white_check_mark: vkublytskyi<br/>:white_check_mark: andrew-garside-temando<br/>:white_check_mark: ishakhsuvarov<br/>:white_check_mark: dmanners<br/>:white_check_mark: RomaKis<br/>:x: rossbrandon<br/>:x: olysenko<br/>:x: AlexMaxHorkun<br/>:x: aakimov<br/>:x: odubovyk<br/>:x: interio<br/>:x: dvoskoboinikov

sidolov commented:
Described issue resolved in https://github.com/magento/magento2/pull/12167
